src/main/java/frc/robot/constants/AbstractConstants.java

```diff
@@ -166,6 +226,22 @@ public int getGyroId() {
      */
     public abstract Rotation2d[] getSwerveModuleEncoderOffsets();
 
+    public SwirveModuleHardwareIds getSwirveModuleHardwareIdsFrontLeft() {
+        return new SwirveModuleHardwareIds(0, 1, 2);
+    }
+
+    public SwirveModuleHardwareIds getSwirveModuleHardwareIdsFrontRight() {
+        return new SwirveModuleHardwareIds(3, 4, 5);
+    }
+
+    public SwirveModuleHardwareIds getSwirveModuleHardwareIdsBackLeft() {
+        return new SwirveModuleHardwareIds(9, 10, 11);
+    }
+
+    public SwirveModuleHardwareIds getSwirveModuleHardwareIdsBackRight() {
+        return new SwirveModuleHardwareIds(6, 7, 8);
+    }
```

Comment:
This is a pretty consistent approach to handling the constants, but we lose the ability to easily see the complete list of motor IDs (which must be unique). My preference would be for the software to insist on the IDs it requires, because the robot is delivered without motor IDs (all zeros or leftover values from old robots, with possible conflicts) so we end up setting them anyway.

Reply:
I Moved All CAN Bus ID & roboRIO Port Constants to AbstractConstantsand added uniqueness checks.
Let me know what you think.